src/parsers/forecast_parser.jl

```diff
@@ -69,36 +70,55 @@ function _forecast_csv_parser!(sys::System, forecast_infos::ForecastInfos, resol
             continue
         end
 
-        if isconcretetype(forecast.category)
-            component = get_component(forecast.category, sys, forecast.component_name)
+        if forecast.component isa LoadZones
+            uuids = Set([get_uuid(x) for x in forecast.component.buses])
+            forecast_components = [load for load in get_components(ElectricLoad, sys)
+                                   if get_bus(load) |> get_uuid in uuids]
         else
-            components = get_components_by_name(forecast.category, sys,
-                                                forecast.component_name)
-            if length(components) == 0
-                @error("Did not find component for forecast", forecast.component_name,
-                       forecast.category, forecast.file_path)
-                continue
-            elseif length(components) == 1
-                component = components[1]
-            else
-                msg = "Found duplicate names type=$(forecast.category) " *
-                      "name=$(forecast.component_name)"
-                throw(DataFormatError(msg))
-            end
+            forecast_components = [forecast.component]
         end
 
+        per_unit_conversions = Set()
         forecasts = Vector{Forecast}()
-        forecast_components = component isa LoadZones ? component.buses : [component]
-        for component_ in forecast_components
-            timeseries = forecast.data[Symbol(forecast.component_name)]
-            forecast_ = Deterministic(component_, forecast.label, timeseries)
+        for component in forecast_components
+            sym = Symbol(get_name(forecast.component))
+            timeseries = forecast.data[sym]
+            if sym in per_unit_conversions
+                @assert forecast.per_unit
+            elseif forecast.per_unit
+                # PERF
```

Comment:
Note this.  It is unfortunate.  If we used DataFrames instead then we could avoid copies.

Reply:
DataFrames instead of TimeSeriesArrays?

Reply:
Yes.  I'm not advocating for a switch.  I still like the choice of TimeArrays for this purpose.